includes/api/class-edd-api.php

```diff
@@ -1245,72 +1248,96 @@ public function get_stats( $args = array() ) {
 					$start_date = $dates['year'] . '-' . $dates['m_start'] . '-' . $dates['day_start'];
 					$end_date   = $dates['year_end'] . '-' . $dates['m_end'] . '-' . $dates['day_end'];
 
-					$stats = EDD()->payment_stats->get_sales_by_range( 'other', true, $start_date, $end_date );
+					// Force the data for the reports API.
+					$_GET['filter_from'] = $start_date;
+					$_GET['filter_to']   = $end_date;
+					$_GET['range']       = 'other';
 
-					foreach ( $stats as $sale ) {
-						$key                    = $sale['y'] . $sale['m'] . $sale['d'];
-						$sales['sales'][ $key ] = (int) $sale['count'];
-					}
+					$stats   = new EDD\Stats();
+					$dates   = EDD\Reports\parse_dates_for_range();
+
+					$total_sales = $stats->get_order_count(
+						array(
+							'start'        => $dates['start']->format( 'Y-m-d H:i:s' ),
+							'end'          => $dates['end']->format( 'Y-m-d H:i:s' ),
+							'revenue_type' => 'net',
```

Comment:
This needs to be moved to where the class is instantiated in line 1256, I think, due to #9347.